lib/signatures.ts

```diff
@@ -1,25 +1,34 @@
 import { promises } from 'fs';
-import { SignatureHashAlgorithm, SignatureResult } from './types';
+import { SignatureFormat, SignatureResult } from './types';
 import { getDubHashSignature } from './dubhash';
 import { getUhashSignature } from './uhash';
 import pMap = require('p-map');
 import { join } from 'path';
 
 const { readFile } = promises;
 
+type Signer = (path: string, content: Buffer) => SignatureResult | null;
```

Comment:
Getting rid of the abstraction as it will not be needed see https://github.com/snyk/snyk-cpp-plugin/pull/44#discussion_r725881687

Will change the need for Signer